### PR TITLE
[MINOR] feat(spark): Support reporting error messages from client

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Support reporting more error messages rather than only `Fail to send the block`.

### Why are the changes needed?

Let users know the error code so users can solve issues by themselves.

### Does this PR introduce _any_ user-facing change?

No, just output more error messages.

### How was this patch tested?

In our env.

### CODE REVIEW

File: client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java [hunk 1]
```diff
@@ -513,23 +513,32 @@ private void checkDataIfAnyFailure() {
...
 
-  private boolean hasAnyBlockFailure() {
+  private String getFirstBlockFailure() {
     Set<Long> failedBlockIds = shuffleManager.getFailedBlockIds(taskId);
     if (!failedBlockIds.isEmpty()) {
+      List<TrackingBlockStatus> trackingBlockStatues =
+          shuffleManager
+              .getBlockIdsFailedSendTracker(taskId)
+              .getFailedBlockStatus(failedBlockIds.iterator().next());
+      String errorMsg = "Default Error Message";
```
Comment (rickyma) [inline client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java:530]: Could we use a static final variable to replace this?

**Comment by maobaolong** [comment]
@rickyma Thanks for your review and suggestion, I introduced a constant, PTAL

**Review by rickyma — APPROVED** [review]
LGTM

**Comment by rickyma** [comment]
Merged to master. Thanks for your contribution @maobaolong

**Pull request merged**